app/tests/src/test_login.py

```diff
@@ -1,34 +1,24 @@
-import os
-
 import chainlit.config
-import src.shared
+from src.app_config import app_config
 from src.login import login_callback, require_login
 
 
 def test_require_login_no_password(monkeypatch):
-    if "GLOBAL_PASSWORD" in os.environ:
-        monkeypatch.delenv("GLOBAL_PASSWORD")
-
-    # Rebuild AppConfig with new environment variables
-    src.shared.get_app_config.cache_clear()
-
+    app_config.global_password = None
```

Comment:
Set the `global_password` directly and avoid having to `cache_clear()`. This assumes that `PydanticBaseEnvConfig` correctly sets `global_password` from the env variable, which is a valid assumption since the point of this test is `login.py`.